core/src/main/scala/org/apache/spark/scheduler/dynalloc/ExecutorMonitor.scala

```diff
@@ -137,6 +138,17 @@ private[spark] class ExecutorMonitor(
     timedOutExecs.sortBy(_._1)
   }
 
+  /** Return executors can be removed constrained by [[decommissionMaxRatio]] */
+  def executorsCanBeRemoved(): Seq[(String, Int)] = {
+    if (hasStorageDecommission()) {
+      val maxExecutorsCanBeRemoved = math.floor(executorCount * decommissionMaxRatio
+        - decommissioningCount).toInt
```

Comment:
If we are going with a ratio, make this a ceil and not floor - to prevent `maxExecutorsCanBeRemoved` from being `0`